security/crypto/bls12/bls12.go

```diff
@@ -350,75 +343,85 @@ func (bls *bls12Base) Combine(signatures ...hotstuff.QuorumSignature) (combined
 			}
 			g2.Add(&agg, &agg, &sig2.sig)
 		} else {
-			bls.logger.Panicf("cannot combine incompatible signature type %T (expected %T)", sig1, sig2)
+			return nil, fmt.Errorf("cannot combine incompatible signature type %T (expected %T)", sig1, sig2)
```

Comment:
Maybe these types of errors can be made a `FatalError` since we actually want to log with a `Panic` for these things... I think these panics here mean that there is something wrong with the configuration, and the user should fix it.

But we want to return an error here rather than log things at this level, since we want to have tests for this stuff, and testing for errors is easier than testing for errors in logs.